lib/nats.js

```diff
@@ -273,11 +277,17 @@ Client.prototype.selectServer = function() {
   client.url = server.url;
   if ('auth' in server.url && !!server.url.auth) {
     var auth = server.url.auth.split(':');
-    if (client.options.user === undefined) {
-      client.user = auth[0];
-    }
-    if (client.options.pass === undefined) {
-      client.pass = auth[1];
+    if (auth.length !== 1) {
+      if (client.options.user === undefined) {
+        client.user = auth[0];
+      }
+      if (client.options.pass === undefined) {
+        client.pass = auth[1];
+      }
+    } else {
+      if (client.options.token === undefined) {
+        client.token = auth[0];
```

Comment:
I am not sure if you are advising a change or not here. If you are can you clarify more?